tests/integration/integration_test_runner.sh

```diff
@@ -0,0 +1,223 @@
+rm -f last-diff-result.log
+source tests/integration/env.test && ./main.sh
+
+
+### Asserts that diff contains the right thing
+printf "%s\n" "" "    [TEST] ASSERT" ""
+if grep -q 'export WEAVER_WORKER_IMAGE="pavics/weaver:1.13.3-worker"' "last-diff-result.log"; then
+    printf "${GREEN}[INFO] [bump_weaver-worker_to_1.13.3-worker] The commit content looks good"
+    echo
+    echo
+    cat last-diff-result.log
+    printf "${NC}"
+    ((SUCCESS_COUNT++))
+else
+    printf "${RED}[ERROR] [bump_weaver-worker_to_1.13.3-worker] wrong commit content."
+    echo
+    echo
+    cat last-diff-result.log
+    printf "${NC}"
+    ((FAILURE_COUNT++))
+fi
+
+
+# update ALL images
+printf "%s\n" "" "    [INFO] Pushing all tags to DockerHub" ""
+printf "${BLUE}"
+curl -s -XPOST localhost:5000/pavics/weaver/1.13.4-worker
+curl -s -XPOST localhost:5000/pavics/weaver/1.13.4-manager
+curl -s -XPOST localhost:5000/pavics/weaver/1.13.4
+curl -s -XPOST localhost:5000/birdhouse/finch/version-0.5.5
+printf "${NC}"
+PUSHED_TAGS_COUNT=4
+echo
+
+# count number of updated image tags
+UPDATED_IMAGE_COUNTER=-1
+exitCode=100
+
+while [[ $exitCode -eq 100 ]]
+do
+    # run updater CLI
+    rm -f last-diff-result.log
+    source tests/integration/env.test && ./main.sh
+
+    exitCode=$?
+    UPDATED_IMAGE_COUNTER=$((UPDATED_IMAGE_COUNTER+1))
+    cat $DATA_DIR/last-update-result.log >> $DATA_DIR/updated-images-list.log
+done
+
+
+### Asserts that correct number of images have been updated
+printf "%s\n" "" "    [TEST] ASSERT" ""
+if [[ $UPDATED_IMAGE_COUNTER -eq $PUSHED_TAGS_COUNT ]]; then
+    printf "${GREEN}[INFO] All image tags push created a PR have triggered the image updater."
+    echo
+    echo
+    cat $DATA_DIR/updated-images-list.log
+    printf "${NC}"
+    ((SUCCESS_COUNT++))
+else
+    printf "${RED}[ERROR] Wrong number of image tag resulting in PR. Expecting [$PUSHED_TAGS_COUNT], got [$UPDATED_IMAGE_COUNTER]."
+    echo
+    echo
+    cat $DATA_DIR/updated-images-list.log
+    printf "${NC}"
+    ((FAILURE_COUNT++))
+fi
+
+
+# update an image
+printf "%s\n" "" "    [INFO] Pushing weaver tag to DockerHub" ""
+printf "${BLUE}"
+curl -s -XPOST localhost:5000/pavics/weaver/1.13.5-worker
+printf "${NC}"
+
+# run updater CLI
+printf "%s\n" "" "    [TEST] Running updater - need to create a PR for [bump_weaver-worker_to_1.13.4-worker]" ""
+rm -f last-diff-result.log
+source tests/integration/env.test && ./main.sh
+
+
+### Assert
+printf "%s\n" "" "    [TEST] ASSERT" ""
+if grep -q '1.13.5-worker' "$DATA_DIR/last-update-result.log"; then
+    printf "${GREEN}[INFO] [bump_weaver-worker_to_1.13.5-worker] 'last-update-result.log' looks good"
+    echo
+    echo
+    cat $DATA_DIR/last-update-result.log
+    printf "${NC}"
+    ((SUCCESS_COUNT++))
+else
+    printf "${RED}[ERROR] [bump_weaver-worker_to_1.13.5-worker] wrong 'last-update-result.log'."
+    echo
+    echo
+    cat $DATA_DIR/last-update-result.log
+    printf "${NC}"
+    ((FAILURE_COUNT++))
+fi
```

Comment:
Nice integration test.  A few duplicate code block, worth to factor out to a function for re-use.

Reply:
Yes, same thing I was thinking while copy-pasting the log and assertion related logic. 
I'll put this in the todo list, since readability would greatly be improved.